### PR TITLE
Sync to intl-duration-format PR 119

### PR DESCRIPTION
**Pull request by FrankYFTang**
Sync to https://github.com/tc39/proposal-intl-duration-format/pull/119
Use "unit" type for listFormat

### CODE REVIEW

**Comment by FrankYFTang** [comment]
@ryzokuken @romulocintra @sffc

**Review by romulocintra — APPROVED** [review]

**Comment by ptomato** [comment]
@FrankYFTang Could you address the comments from @trflynn89 or say why you disagree?

**Comment by trflynn89** [comment]
FWIW my comment was partially addressed in #3640 - that PR (now merged) ended up with the fix for `basic-format-en.js`, so that can be dropped from this PR.

I think all that remains is addressing the commas in the `narrow` test in `style-options-en.js`.

**Comment by FrankYFTang** [comment]
I think I need to update this PR again after some other spec PRs and test262 PRs merged. Please wait. sorry

**Comment by FrankYFTang** [comment]
@ryzokuken @sffc I believe now this PR is updated to the latest spec. @romulocintra PTAL

**Review by romulocintra — APPROVED** [review]

**Review by ryzokuken — DISMISSED** [review]

File: test/intl402/DurationFormat/prototype/format/style-options-en.js [hunk 1]
```diff
@@ -9,10 +9,10 @@ features: [Intl.DurationFormat]
 ---*/
 
 const testData = {
-  "long" : "1 year, 2 months, 3 weeks, 3 days, 4 hours, 5 minutes, 6 seconds, 7 milliseconds, 8 microseconds, and 9 nanoseconds",
-  "short": "1 yr, 2 mths, 3 wks, 3 days, 4 hr, 5 min, 6 sec, 7 ms, 8 μs, and 9 ns",
-  "narrow":"1y, 2m, 3w, 3d, 4h, 5m, 6s, 7ms, 8μs, and 9ns",
-  "digital":"1y, 2m, 3w, 3d, and 4:05:06.007",
+  "long" : "1 year, 2 months, 3 weeks, 3 days, 4 hours, 5 minutes, 6 seconds, 7 milliseconds, 8 microseconds, 9 nanoseconds",
```
Comment (ptomato) [inline test/intl402/DurationFormat/prototype/format/style-options-en.js:12]: Is this line correct? By my reading of https://tc39.es/proposal-intl-duration-format/#sec-properties-of-intl-durationformat-constructor the list formatter is constructed with `{ style: 'long' }`, which produces a list with the format "A, B, and C"

Comment (trflynn89) [reply test/intl402/DurationFormat/prototype/format/style-options-en.js:12]: `DurationFormat` also sets the option `{ type : 'unit' }` when it constructs the `ListFormat` object. The CLDR does not use "and" in the list pattern for units:
https://github.com/unicode-org/cldr/blob/main/common/main/en.xml#L9107-L9112

Comment (ptomato) [reply test/intl402/DurationFormat/prototype/format/style-options-en.js:12]: ah, got it, thanks.

**Review by ptomato — APPROVED** [review]

**Review by ptomato — APPROVED** [review]

**Pull request merged**